packages/legend-query-builder/src/stores/shared/LambdaEditorState.ts

```diff
@@ -69,6 +70,13 @@ export abstract class LambdaEditorState {
     this.setParserError(undefined);
   }
 
+  keepError(): void {
```

Comment:
this method seems a bit confusing, I think it's better to have an option to alter the behavior of `clearErrors` instead, e.g.

```ts
clearErrors(options: { preserveCompilationError?: boolean })
```